### ISSUE TITLE
Every time pod is restarting I lost all data due to wrong NFS-Permissions set

### ISSUE THREAD

**Issue by diegoavitorino** [louislam/uptime-kuma#4847]
### ⚠️ Please verify that this question has NOT been raised before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/uptime-kuma/security/policy)

### 📝 Describe your problem

Every time pod is restarting I lost all data and back to setup a new account, even pv and pvc configured

### 📝 Error Message(s) or Log

```
2024-06-11T17:19:35Z [SERVER] INFO: JWT secret is not found, generate one.
2024-06-11T17:19:36Z [SERVER] INFO: Stored JWT secret into database
2024-06-11T17:19:36Z [SERVER] INFO: No user, need setup
2024-06-11T17:19:36Z [SERVER] INFO: Adding route
2024-06-11T17:19:37Z [SERVER] INFO: Adding socket handler
2024-06-11T17:19:37Z [SERVER] INFO: Init the server
2024-06-11T17:19:37Z [SERVER] INFO: Listening on 3001
2024-06-11T17:19:37Z [SERVICES] INFO: Starting nscd
2024-06-11T17:19:53Z [SOCKET] INFO: New polling connection, IP = 100.64.31.81
2024-06-11T17:19:53Z [SERVER] INFO: Redirect to setup page
2024-06-11T17:19:53Z [AUTH] INFO: Login by token. IP=100.64.31.81
2024-06-11T17:19:53Z [AUTH] ERROR: Invalid token. IP=100.64.31.81
2024-06-11T17:19:53Z [AUTH] ERROR: invalid signature
2024-06-11T17:19:53Z [RATE-LIMIT] INFO: remaining requests: 19
```

### 🐻 Uptime-Kuma Version

latest

### 💻 Operating System and Arch

Kubernetes

### 🌐 Browser

Google Chrome/Edge

### 🖥️ Deployment Environment

- Runtime: 
- Database: 
- Filesystem used to store the database on: 
- number of monitors:

**Comment by CommanderStorm** [comment]
> even pv and pvc configured

> Operating System and Arch: Kubernetes

Could you be a bit more specific about your configuration? It is really hard to debug without having the data.

You are likely missing the `-v uptime-kuma:/app/data` equivalent part of the docker run command or have set the data directory to not be in the PV/PVC

Are you using the unofficial  helm chart or have you written your own?
https://github.com/louislam/uptime-kuma/wiki/%F0%9F%94%A7-How-to-Install#%EF%B8%8F-openshift-4-and-kubernetes-helm-3-chart-unofficial

**Comment by diegoavitorino** [comment]
Hi, @CommanderStorm 
![image](https://github.com/louislam/uptime-kuma/assets/44095125/0affa169-e22e-4f56-8702-1b6643d5745b)

I've configured on my own, but I can also follow https://github.com/k3rnelpan1c-dev/uptime-kuma-helm as you inform

apiVersion: v1
kind: Namespace
metadata:
  name: uptime-kuma
---
apiVersion: v1
kind: Service
metadata:
  name: uptime-kuma-tcp
  namespace: uptime-kuma
spec:
  ports:
  - name: web-ui
    protocol: TCP
    port: 3001
    targetPort: 3001
  selector:
    app: uptime-kuma
---
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: uptime-kuma-pvc
  namespace: uptime-kuma
spec:
  accessModes:
    - ReadWriteMany
  resources:
    requests:
      storage: 1Gi
  storageClassName: efs
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: uptime-kuma
  namespace: uptime-kuma
spec:  
  selector:
    matchLabels:
      app: uptime-kuma
  replicas: 1
  strategy:
    type: RollingUpdate
    rollingUpdate:
      maxSurge: 1
      maxUnavailable: 1
  template:
    metadata:
      labels:
        app: uptime-kuma
    spec:   
      containers:    
      - name: uptime-kuma
        image: louislam/uptime-kuma:latest
        imagePullPolicy: IfNotPresent     
        env:
        # only need to set PUID and PGUI because of NFS server
        - name: PUID
          value: "1000"
        - name: PGID
          value: "1000"         
        ports:
        - containerPort: 3001
          name: web-ui
        resources:
          limits:
            cpu: 200m
            memory: 512Mi
          requests:
            cpu: 50m
            memory: 128Mi
        livenessProbe:
          tcpSocket:
            port: web-ui
          initialDelaySeconds: 60
          periodSeconds: 10
        readinessProbe:
          httpGet:
            scheme: HTTP
            path: /
            port: web-ui
          initialDelaySeconds: 30
          periodSeconds: 10                      
        volumeMounts:
        - name: data
          mountPath: /app/data                    
      volumes:
      - name: data
        persistentVolumeClaim:
          claimName: uptime-kuma-pvc

**Comment by CommanderStorm** [comment]
So that does not look quite wrong, but also not quite right.
I can't see a real reason why this would misbehave as reported. Have you checked that
- it is applied and that [`DATA_DIR`](https://github.com/louislam/uptime-kuma/wiki/Environment-Variables#server-environment-variables) is not set to a different path?
- Have you ensured that the home directory of User `1000`?
  `DATA_DIR` defaults to `./data/` => if we store to a different path...

So a few general things:

- About those requests and limits, please re-read that you are entirely sure what you are doing.
 Setting a CPU-limit is almost never what you want: https://home.robusta.dev/blog/stop-using-cpu-limits
  
  The memory might be too low depending what monitor you want to run (all except real browser are likely fine).
- ```yaml
  storage: 1Gi
  ```
  
  Please see #4500 and what your deployment will be in terms of retention and monitor count. In the `v1.x`-series, we are somewhat wasteful with storage.
- ```yaml
  accessModes:
  - ReadWriteMany
  ```
  We don't support this access mode. Please use `ReadWriteOnce` instead.
- ```yaml
  image: louislam/uptime-kuma:latest
  imagePullPolicy: IfNotPresent
  ```
  
  If you fix your image via `IfNotPresent`, you should also fix the version. Otherwise you will have a bad time debugging stuff if shit hits the fan. (as you won't reliably know what you are running)
- Please consider at least pinning yourself at `louislam/uptime-kuma:1` instead of `:latest`.


<details><summary>and there are a few things which you did not configure which might not be intentional</summary>
<p>

- you can tighten your `securityContext` and several other settings
- you are missing `nodeSelector`, `affinity`, `tolerations`
- You have not configured a `serviceMonitor`, `ingress`

</p>
</details>

**Comment by diegoavitorino** [comment]
I check options, now I'm facing the error bellow:

chown: changing ownership of '/app/data': Operation not permitted

Follow my yamls

apiVersion: v1
kind: Service
metadata:
  name: uptime-kuma-tcp
  namespace: uptime
spec:
  ports:
  - name: web-ui
    protocol: TCP
    port: 3001
    targetPort: 3001
  selector:
    app: uptime-kuma
---
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: uptime-kuma-pvc
  namespace: uptime
spec:
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 1Gi
  storageClassName: efs
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: uptime-kuma
  namespace: uptime
spec:  
  selector:
    matchLabels:
      app: uptime-kuma
  replicas: 1
  strategy:
    type: RollingUpdate
    rollingUpdate:
      maxSurge: 1
      maxUnavailable: 1
  template:
    metadata:
      labels:
        app: uptime-kuma
    spec:   
      # securityContext:
      #   runAsUser: 1000
      #   runAsGroup: 1000
      #   fsGroup: 1000    
      containers:    
      - name: uptime-kuma
        image: louislam/uptime-kuma:1
        imagePullPolicy: IfNotPresent     
        env:
        # only need to set PUID and PGUI because of NFS server
        - name: PUID
          value: "1000"
        - name: PGID
          value: "1000"         
        ports:
        - containerPort: 3001
          name: web-ui
        livenessProbe:
          tcpSocket:
            port: web-ui
          initialDelaySeconds: 60
          periodSeconds: 10
        readinessProbe:
          httpGet:
            scheme: HTTP
            path: /
            port: web-ui
          initialDelaySeconds: 30
          periodSeconds: 10                          
        volumeMounts:
        - name: data
          mountPath: /app/data                    
      volumes:
      - name: data
        persistentVolumeClaim:
          claimName: uptime-kuma-pvc

**Comment by CommanderStorm** [comment]
What file-permissions does `PUID=1000` have?
Is it able to access `/app/data`?

**Comment by diegoavitorino** [comment]
The pod not even start, is there possible to give access to user 1000?

I tried:

    spec:   
      securityContext:
        runAsUser: 1000 
        runAsGroup: 1000
        fsGroup: 1000        
      containers:    
      - name: uptime-kuma
        image: louislam/uptime-kuma:1
        imagePullPolicy: IfNotPresent     
        env:
        # only need to set PUID and PGUI because of NFS server
        - name: PUID
          value: "1000"
        - name: PGID
          value: "1000"         
        ports:
        - containerPort: 3001
          name: web-ui
        livenessProbe:
          tcpSocket:
            port: web-ui
          initialDelaySeconds: 60
          periodSeconds: 10
        readinessProbe:
          httpGet:
            scheme: HTTP
            path: /
            port: web-ui
          initialDelaySeconds: 30
          periodSeconds: 10
        securityContext:
          privileged: true                         
        volumeMounts:
        - name: data
          mountPath: /app/data                    
      volumes:
      - name: data
        persistentVolumeClaim:
          claimName: uptime-kuma-pvc

**Comment by diegoavitorino** [comment]
Fy, I'm tryng to use EFS storage class

**Comment by CommanderStorm** [comment]
> is there possible to give access to user `PUID=1000`

If the user `1000` and group `1000` has the right permissions, to start/stop system services and to access the allocated file system, you should not have a problem.

=> what permission does the file system have? Have you cleanly deleted the PVC and re-added it? (k8s is kind of a bitch and sticks with old configurations for FS-stuff sometimes)

**Comment by CommanderStorm** [comment]
Yes, the support for `PUID` and `GUID` is somewhat spotty. See #4500 for further details.

**Comment by diegoavitorino** [comment]
As I can see EFS/NFS is not supported, everything I created on data folder, after delete pod was lost

**Comment by diegoavitorino** [comment]
I was able through this post https://github.com/louislam/uptime-kuma/issues/2668

"While testing NFS on K8s, I ran into the typical chmod issue due to the fact that the container cannot change the permissions of the NFS share.

This is usually solved by using a sub-folder so I tried using /app/data/sub and setting DATA_DIR accordingly. This did not work.

Instead, I could get the container started using:

NFS mapped to /data
DATA_DIR: /data/sub"
